### DISCUSSION TITLE
Is the inspector compatible with SSR?

### DISCUSSION THREAD

**Discussion by dminkovsky** [statelyai/xstate#1518] [Q&A] [answered]
I'm trying to run @xstate/inspect with SSR (Next.js) but it won't let me register it early in my application, at the top level, because `inspect()` accesses `window`, which is not available in Node. 

I am using React, and with React the usual workaround for this kind of thing is calling such a function in `useEffect`, which only runs in the client and can therefore safely access `window`. But for this use-case, this seems to be too late. The inspector pops up, but just shows "waiting for connection." Am I missing some way to get this working?

Thanks!

**Comment by dminkovsky** [comment, marked as answer]
Looks like this was mentioned on inspector thread: https://github.com/davidkpiano/xstate/discussions/1430#discussioncomment-87284. @RafalFilipek did you get it working with SSR?

**Reply by RafalFilipek** [reply]
Hi,

```
 try {
        inspect({
          url: "https://statecharts.io/inspect",
          iframe: false,
        });
        // eslint-disable-next-line no-empty
      } catch (e) {}
```

;)

**Reply by dminkovsky** [reply]
Wow that's so nice, that never occurred to me, thanks.
